packages/perseus/src/widgets/categorizer/validate-categorizer.ts

```diff
@@ -0,0 +1,28 @@
+import type {PerseusStrings} from "../../strings";
+import type {PerseusScore} from "../../types";
+import type {
+    PerseusCategorizerRubric,
+    PerseusCategorizerUserInput,
+} from "../../validation.types";
+
+function validateCategorizer(
+    userInput: PerseusCategorizerUserInput,
+    rubric: PerseusCategorizerRubric,
+    strings: PerseusStrings,
+): Extract<PerseusScore, {type: "invalid"}> | null {
+    let completed = true;
+    rubric.items.forEach((_, i) => {
+        if (userInput.values[i] == null) {
+            completed = false;
+        }
+    });
+    if (!completed) {
+        return {
+            type: "invalid",
+            message: strings.invalidSelection,
+        };
+    }
+    return null;
```

Comment:
I know this is just moved code, but what about something like:

```suggestion
    let incomplete = rubric.items.some((_, i) => userInput.values[i] == null);
    
    if (incomplete) {
        return {
            type: "invalid",
            message: strings.invalidSelection,
        };
    }
    
    return null;
```

Reply:
I'm all for simplifying. I'll try to simplify the validation logic in the future if it seems like a straightforward update. Was trying to leave as close to original as possible as we've mentioned not changing the logic, but I think just simplifying what's there is probably fine. Updated :)